File: src/test/java/interview/reader/ConsoleInputReaderTest.java
package interview.reader;

import org.junit.jupiter.api.Test;

import java.io.ByteArrayInputStream;
import java.io.InputStream;

import static org.junit.jupiter.api.Assertions.assertEquals;

/**
 * @author : Xenofon Zinoviou
 */
class ConsoleInputReaderTest {

    private ConsoleInputReader testClass;

    @Test
    void test_GivenValidInput_thenReturnInputAsString() {
        String text = "Hello World";
        testClass = new ConsoleInputReader(getInputStream(text));

        final String result = testClass.readInput();

        assertEquals(text, result);
    }

    @Test
    void test_GivenEmptyInput_thenReturnInputAsEmptyString() {
        String text = "";
        testClass = new ConsoleInputReader(getInputStream(text));

        final String result = testClass.readInput();

        assertEquals(text, result);
    }

    private InputStream getInputStream(String text) {
        return new ByteArrayInputStream(text.getBytes());
    }
}

File: src/test/kotlin/inputReader/ConsoleInputReaderTest.kt
package inputReader

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class ConsoleInputReaderTest {

    @Test
    fun readsPrintTextAndReadsInputThenReturnsInput() {
        // given
        var printedValue: String? = null
        val valueForInput = "test input text"
        val printer = { value: String -> printedValue = value }
        val reader = { valueForInput }
        val consoleInputReader = ConsoleInputReader(reader, printer)

        // when
        val result = consoleInputReader.read()

        // then
        assertEquals(ConsoleInputReader.entryMessage, printedValue)
        assertEquals(valueForInput, result)
    }
}